File: src/main/java/pl/training/workshop/m3/s06_yagni/step1/VipRule.java
package pl.training.workshop.m3.s06_yagni.step1;

import java.math.BigDecimal;
import java.util.Map;

/** Miejsce VIP +10.00. */
public final class VipRule implements PricingRule {
    @Override
    public boolean appliesTo(Map<String, Object> context) {
        return (int) context.get("row") >= (int) context.get("vipFromRow");
    }

    @Override
    public BigDecimal apply(Map<String, Object> context, BigDecimal price) {
        return price.add(new BigDecimal("10.00"));
    }
}

File: csharp/src/Training.Workshop/M3/S06Yagni/Step1/VipRule.cs
namespace Training.Workshop.M3.S06Yagni.Step1;

/// <summary>Miejsce VIP +10.00.</summary>
public sealed class VipRule : IPricingRule
{
    public bool AppliesTo(IDictionary<string, object> context)
    {
        return (int)context["row"] >= (int)context["vipFromRow"];
    }

    public decimal Apply(IDictionary<string, object> context, decimal price)
    {
        return price + 10.00m;
    }
}
